### PR TITLE
Feat save data from daily medication manager

### LINKED ISSUE #48: Interface with DailyMedicationManager to save to text file
For the v1.0 implementation, we will just have a file called `today.txt` under the folder `data/dailymed`
Will take in and return `List<String>` from and to the `DailyMedicationManager`

Does not check for the validity of the data, the DailyMedicationManager will handle that.

### PR DESCRIPTION
**Pull request by annoy-o-mus**
# Checklist for feature implementation and modification
Closes #48 (Requires #56 to be merged first)
- [x] Added relevant milestone (under milestone tab on the right)
- Implementation
  - [x] Sources (if any) credited under both relevant section(s) and Developer Guide (Acknowledgement)
  - [ ] JUnit Test(s)
- Documentation
  - [x] JavaDoc for classes and methods
  - Developer Guide
    - [x] Design (Can add as a todo statement for later on)
    - [x] Implementation (Can add as a todo statement for later on)
    - [ ] Relevant User Story present
    - [ ] Non-functional requirement(s) that the feature addresses (if any)
    - [ ] Glossary of the (technical) term(s) used (if any)
    - [ ] Specialised instructions for manual testing (if any)
  - User Guide
    - [ ] Description and usage of feature (if any)
  - Contributions (under `/docs/team/*.md`)
    - [x] Code/Feature/Enhancement contribution
    - [ ] Developer Guide contribution
    - [ ] User Guide contribution
    - [ ] Misc and team-based admin tasks (if any)

# Additional Information
To only be reviewed after
- [ ] #56 
is merged.

### CODE REVIEW

File: src/main/java/meditracker/storage/FileReaderWriter.java [hunk 1]
```diff
@@ -119,4 +130,58 @@ public static void saveMediTrackerData(MedicationManager medManager) {
...
     }
+
+    /**
+     * Loads the daily medication information from a fixed file data/dailymed/today.txt.
+     * The loading functionality will be improved in v2.0.
+
+     * @param dailyMedData A list of type String for the daily medication data.
+     * @throws FileReadWriteException if there is an issue creating the file.
+     */
+    public static void saveDailyMedicationData(List<String> dailyMedData) throws FileReadWriteException {
```
Comment (T0nyLin) [inline src/main/java/meditracker/storage/FileReaderWriter.java:141]: is the JavaDoc (line 135) and saveDailyMedicationData() name swapped

File: src/main/java/meditracker/storage/FileReaderWriter.java [hunk 2]
```diff
@@ -119,4 +130,58 @@ public static void saveMediTrackerData(MedicationManager medManager) {
...
+            }
+            f.flush();
+            f.close();
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Saves the daily medication information to a fixed file data/dailymed/today.txt.
```
Comment (T0nyLin) [inline src/main/java/meditracker/storage/FileReaderWriter.java:172]: Just like the comment above, should this be "Loads the ..." as the method is loadDailyMedicationData() ?

**Review by T0nyLin — CHANGES_REQUESTED** [review]
Code looks clean and with proper JavaDoc. Just a few clarification below in the comments.

Can I ask where are the methods "saveDailyMedicationData()" and "loadDailyMedicationData()" being called?

File: src/main/java/meditracker/storage/FileReaderWriter.java (hunk 1 above)
Comment (annoy-o-mus) [reply src/main/java/meditracker/storage/FileReaderWriter.java:142]: Ah yes, thanks for the catch. Must have swapped the javadoc for the 2 functions around during committing. Fixed in cbca7dfa7748336954d21dfcbd4d7911f48f34c7

File: src/main/java/meditracker/storage/FileReaderWriter.java (hunk 2 above)
Comment (annoy-o-mus) [reply src/main/java/meditracker/storage/FileReaderWriter.java]: See above comment

**Comment by annoy-o-mus** [comment]
Both `saveDailyMedicationData()` and `loadDailyMedicationData()` are static and can be called directly via `FileReaderWriter`.

For the saving, it is on demand. meaning the saving has to be called explicitly in code, but it can be called anywhere in the code so long as you have a `List<String>` to save.

As for the loading, it is supposed to be called in `MediTracker` when the program first starts up. Since I have merged in the recent changes, I can introduce the addition in MediTracker for your review. (Implemented in xxx)

We should really make all the managers static sometime.

**Review by T0nyLin — APPROVED** [review]
LGTM!

**Pull request merged**